pref-adjust.lua

```diff
@@ -91,16 +136,201 @@ function clearpref_all_dwarves()
+    end
+    if printflag==1 then
+        print("\nTYPE 8 SHAPES:"..list_of_shapes_string) --    printall(list_of_shapes)
+    end
+-- ------------------------------------
+    list_of_poems={} -- Type 9 "likes the words of.."
+    list_of_poems_string = ""
+    vec=df.global.world.poetic_forms.all
+    for k=0,#vec-1 do
+        name=dfhack.TranslateName(vec[k].name,true)
+        list_of_poems[name]=k
+        list_of_poems_string=list_of_poems_string..k..":"..name..","
+    end
+    if printflag==1 then
+        print("\nTYPE 9 POEMS:"..list_of_poems_string) -- printall(list_of_poems)
+    end
+-- ------------------------------------
+    list_of_music={} -- Type 10 "Likes the sound of.."
+    list_of_music_string = ""
+    vec=df.global.world.musical_forms.all
+    for k=0,#vec-1 do
+        name=dfhack.TranslateName(vec[k].name,true)
+        list_of_music[name]=k
+        list_of_music_string=list_of_music_string..k..":"..name..","
+    end
+    if printflag==1 then
+        print("\nTYPE 10 MUSIC:"..list_of_music_string) --    printall(list_of_music)
+    end
+-- ------------------------------------
+    list_of_dances={} -- Type 11
+    list_of_dances_string = ""
+    vec=df.global.world.dance_forms.all
+    for k=0,#vec-1 do
+        name=dfhack.TranslateName(vec[k].name,true)
+        list_of_dances[name]=k
+        list_of_dances_string=list_of_dances_string..k..":"..name..","
+    end
+    if printflag==1 then
+        print("\nTYPE 11 DANCES:"..list_of_dances_string) --    printall(list_of_dances)
+    end
+end -- end func build_all_lists
+
 -- ---------------------------------------------------------------------------
 -- main script operation starts here
 -- ---------------------------------------------------------------------------
-clearpref_all_dwarves()
-adjust_all_dwarves()
+pss_counter=31415926
+utils = require 'utils'
+printflag=0
+build_all_lists(printflag)
+
+local opt = ...
+
+if opt and opt ~= "help" then
+    if opt=="list" then
+        printflag=1
+        build_all_lists(printflag)
+        return
+    end
+    if opt=="clear_one" then
+        local unit = dfhack.gui.getSelectedUnit()
+        if unit==nil then
+            print ("No unit available!  Aborting with extreme prejudice.")
+            return
+        end
+        clear_preferences(unit)
+        prefcount = #(unit.status.current_soul.preferences)
+        print ("After clearing, unit "..dfhack.TranslateName(dfhack.units.getVisibleName(unit)).." has "..prefcount.." preferences")
+        return
+    end
+    if opt=="clear_all" then
+        clearpref_all_dwarves()
+        return
+    end
+    if opt=="goth" then
+        local profile="GOTH"
+        local unit = dfhack.gui.getSelectedUnit()
+        if unit==nil then
+            print ("No unit available!  Aborting with extreme prejudice.")
+        return
+        end
+        clear_preferences(unit)
+        brainwash_unit(unit,profile)
+        return
+    end
+    if opt=="one" then
```

Comment:
This block looks like it's copied directly from above section with one word changed (given that the `return` isn't indented in the same place). Some refactoring to avoid duplicated code is probably in order.

Incidentally, I would use `elseif` instead of a bunch of independent `if`s to make your intentions clearer.